static/js/components/widgets/ResourcePickerDialog.tsx

```diff
@@ -5,42 +5,82 @@ import Dialog from "../Dialog"
 import {
   RESOURCE_TYPE_DOCUMENT,
   RESOURCE_TYPE_IMAGE,
-  RESOURCE_TYPE_VIDEO
+  RESOURCE_TYPE_VIDEO,
+  CONTENT_TYPE_RESOURCE,
+  CONTENT_TYPE_PAGE
 } from "../../constants"
 import ResourcePickerListing from "./ResourcePickerListing"
 import { useDebouncedState } from "../../hooks/state"
 import {
   CKEResourceNodeType,
-  ResourceDialogState,
-  RESOURCE_EMBED
+  ResourceDialogMode,
+  RESOURCE_EMBED,
+  RESOURCE_LINK
 } from "../../lib/ckeditor/plugins/constants"
 import { WebsiteContent } from "../../types/websites"
 
 interface Props {
-  state: ResourceDialogState
+  mode: ResourceDialogMode
+  isOpen: boolean
   closeDialog: () => void
   insertEmbed: (id: string, title: string, variant: CKEResourceNodeType) => void
-  attach: string
 }
 
-const RESOURCE_PICKER_TABS = [
+interface ResourceTabSettings {
+  title: string
+  id: string
+  contentType: "resource" | "page"
+  resourcetype: string | null
+  canEmbed: boolean
+}
+const RESOURCE_PICKER_TABS: ResourceTabSettings[] = [
   {
     title:        "Documents",
-    resourcetype: RESOURCE_TYPE_DOCUMENT
+    id:           RESOURCE_TYPE_DOCUMENT,
+    contentType:  CONTENT_TYPE_RESOURCE,
+    resourcetype: RESOURCE_TYPE_DOCUMENT,
+    canEmbed:     true
   },
   {
     title:        "Videos",
-    resourcetype: RESOURCE_TYPE_VIDEO
+    id:           RESOURCE_TYPE_VIDEO,
+    contentType:  CONTENT_TYPE_RESOURCE,
+    resourcetype: RESOURCE_TYPE_VIDEO,
+    canEmbed:     true
   },
   {
     title:        "Images",
-    resourcetype: RESOURCE_TYPE_IMAGE
+    id:           RESOURCE_TYPE_IMAGE,
+    contentType:  CONTENT_TYPE_RESOURCE,
+    resourcetype: RESOURCE_TYPE_IMAGE,
+    canEmbed:     true
+  },
+  {
+    title:        "Pages",
+    id:           CONTENT_TYPE_PAGE,
+    contentType:  CONTENT_TYPE_PAGE,
+    resourcetype: null,
+    canEmbed:     false
   }
 ]
 
+const modeText = {
+  [RESOURCE_EMBED]: {
+    headerText: "Resources",
+    acceptText: "Embed resource"
+  },
+  [RESOURCE_LINK]: {
+    headerText: "Resources & Pages",
+    acceptText: "Add link"
+  }
+}
+
 export default function ResourcePickerDialog(props: Props): JSX.Element {
-  const { state, closeDialog, insertEmbed, attach } = props
+  const { mode, isOpen, closeDialog, insertEmbed } = props
 
+  const pickerTabs = RESOURCE_PICKER_TABS.filter(
```

Comment:
small naming comment - maybe `activeTabs`, `currentTabs`, something to that effect? up to you, I just always dither over naming things